### PR TITLE
What if Watchman CLI is missing...

### PR DESCRIPTION
**Pull request by alunyov**
Here is an updated version of the `GraphQLWatchmanClient.isAvailable`. (as a fix for https://github.com/facebook/relay/issues/2617)

Now, it's also checking if the `watchman` CLI is available. 

Test plan (Tested with `relay-examples:relay-publish-test`  (https://github.com/relayjs/relay-examples/pull/89)

First of all, I've removed watchman from my machine.
Then,
  - Build the compiler
  - Install updated dependencies relay-examples (yarn)
  - Run the compiler (yarn build)

Compiled without errors.

### CODE REVIEW

File: packages/relay-compiler/core/GraphQLWatchmanClient.js [hunk 1]
```diff
@@ -24,6 +25,14 @@ class GraphQLWatchmanClient {
...
   static isAvailable(): Promise<boolean> {
     return new Promise(resolve => {
+      try {
+        const checkWatchman = execSync('command -v watchman').toString().trim();
+        if (checkWatchman === '') {
+          return resolve(false);
+        }
+      } catch {
+        return resolve(false);
+      }
```
Comment (alloy) [inline packages/relay-compiler/core/GraphQLWatchmanClient.js]: TIL about `command` ! ✨ 

How about using the exit status, though? (And made it async too, although it probably matters very little overall.)

```js
const exec = require("util").promisify(require("child_process").exec)
undefined
> exec("command -v does-not-exist").then(() => console.log("Exists!")).catch(() => console.log("Does not exist!"))
Promise { … }
> Does not exist!
> exec("command -v ls").then(() => console.log("Exists!")).catch(() => console.log("Does not exist!"))
Promise { … }
> Exists!
```

**Comment by alloy** [comment]
Can you clarify if using watchman without `--watch` is faster at yielding files than using `fast-glob` instead? (E.g. does watchman keep a live cached version of the results?)

The amount of faffing with ‘watch’ related options in the code is starting to confuse me a bit, so it would be nice if we could simplify that without a loss in performance.

File: packages/relay-compiler/core/GraphQLWatchmanClient.js (hunk 1 above)
Comment (alunyov) [reply packages/relay-compiler/core/GraphQLWatchmanClient.js]: I just realized that this won't work on windows. I will add something else.

**Comment by alunyov** [comment]
I'm not sure about the performance differences between watchman and fast-glob. But I think at this point, it's fine to keep it as it is.

We've been working recently on the new relay compiler. It's ready, but still has a lot of dependencies on internal tools. And we still are using both versions - but I think once we complete the migration, we will replace an open source version too.

**Pull request closed without merging**